lib/internal/parse_args/parse_args.js

```diff
@@ -0,0 +1,223 @@
+'use strict';
+
+const {
+  ArrayPrototypeConcat,
+  ArrayPrototypeForEach,
+  ArrayPrototypeShift,
+  ArrayPrototypeSlice,
+  ArrayPrototypePush,
+  ObjectPrototypeHasOwnProperty: ObjectHasOwn,
+  ObjectEntries,
+  StringPrototypeCharAt,
+  StringPrototypeIncludes,
+  StringPrototypeIndexOf,
+  StringPrototypeSlice,
+} = primordials;
+
+const {
+  validateArray,
+  validateObject,
+  validateString,
+  validateUnion,
+  validateBoolean,
+} = require('internal/validators');
+
+const {
+  findLongOptionForShort,
+  isLoneLongOption,
+  isLoneShortOption,
+  isLongOptionAndValue,
+  isOptionValue,
+  isShortOptionAndValue,
+  isShortOptionGroup
+} = require('internal/parse_args/utils');
+
+const {
+  codes: {
+    ERR_INVALID_SHORT_OPTION,
+  },
+} = require('internal/errors');
+
+function getMainArgs() {
+  // This function is a placeholder for proposed process.mainArgs.
+  // Work out where to slice process.argv for user supplied arguments.
+
+  // Electron is an interested example, with work-arounds implemented in
+  // Commander and Yargs. Hopefully Electron would support process.mainArgs
+  // itself and render this work-around moot.
+  //
+  // In a bundled Electron app, the user CLI args directly
+  // follow executable. (No special processing required for unbundled.)
+  // 1) process.versions.electron is either set by electron, or undefined
+  //    see https://github.com/electron/electron/blob/master/docs/api/process.md#processversionselectron-readonly
+  // 2) process.defaultApp is undefined in a bundled Electron app, and set
+  //    in an unbundled Electron app
+  //    see https://github.com/electron/electron/blob/master/docs/api/process.md#processversionselectron-readonly
+  // (Not included in tests as hopefully temporary example.)
+  /* c8 ignore next 3 */
+  if (process.versions && process.versions.electron && !process.defaultApp) {
+    return ArrayPrototypeSlice(process.argv, 1);
+  }
+
+  // Check node options for scenarios where user CLI args follow executable.
+  const execArgv = process.execArgv;
+  if (StringPrototypeIncludes(execArgv, '-e') ||
+      StringPrototypeIncludes(execArgv, '--eval') ||
+      StringPrototypeIncludes(execArgv, '-p') ||
+      StringPrototypeIncludes(execArgv, '--print')) {
+    return ArrayPrototypeSlice(process.argv, 1);
+  }
+
+  // Normally first two arguments are executable and script, then CLI arguments
+  return ArrayPrototypeSlice(process.argv, 2);
+}
+
+function storeOptionValue(options, longOption, value, result) {
+  const optionConfig = options[longOption] || {};
```

Comment:
If `??` can be used here instead of `||`, can we do that.

Reply:
This line is changed in https://github.com/pkgjs/parseargs/pull/74/files and the `||` will be gone.